app/src/main/java/com/sma2/sma2/FeatureExtraction/Movement/WalkingFeatureActivity.java

```diff
@@ -127,4 +146,454 @@ private void onButtonBack(){
+
+            if (SignalsLeft.size() > 1) {
+                for (int i = SignalsLeft.size() - 1; i < SignalsLeft.size(); i++) {
+                    path_movement_all_left.add(PATH + SignalsLeft.get(i).getSignalPath());
+                }
+            } else {
+                for (int i = 0; i < SignalsLeft.size(); i++) {
+                    path_movement_all_left.add(PATH + SignalsLeft.get(i).getSignalPath());
+                }
+            }
+        }
+
+
+        if (path_movementLeft == null) {
+            stabVelL = 0;
+        } else {
+            CSVFileReader.Signal TremorSignalaX = FileReader.ReadMovementSignal(path_movementLeft, "aX [m/s^2]");
+            CSVFileReader.Signal TremorSignalaY = FileReader.ReadMovementSignal(path_movementLeft, "aY [m/s^2]");
+            CSVFileReader.Signal TremorSignalaZ = FileReader.ReadMovementSignal(path_movementLeft, "aZ [m/s^2]");
+            stabVelL = MovementProcessor.ComputeRegularity(TremorSignalaX.Signal, TremorSignalaY.Signal, TremorSignalaZ.Signal);
+        }
+        return (float) stabVelL;
+
+    }
+    private float CalculateStabKineticRight() {
+        double stabVelR=0;
+
+        CSVFileReader FileReader = new CSVFileReader(this);
+        MovementProcessing MovementProcessor = new MovementProcessing();
+
+        int IDEx = 27;
+        GetExercises GetEx = new GetExercises(this);
+        String name = GetEx.getNameExercise(IDEx);
+
+
+        SignalDataService signalDataService = new SignalDataService(this);
+        List<SignalDA> SignalsRight = signalDataService.getSignalsbyname(name);
+        if (SignalsRight.size() > 0) {
+            path_movementRight = PATH + SignalsRight.get(SignalsRight.size() - 1).getSignalPath();
+
+            if (SignalsRight.size() > 1) {
+                for (int i = SignalsRight.size() - 1; i < SignalsRight.size(); i++) {
+                    path_movement_all_right.add(PATH + SignalsRight.get(i).getSignalPath());
+                }
+            } else {
+                for (int i = 0; i < SignalsRight.size(); i++) {
+                    path_movement_all_right.add(PATH + SignalsRight.get(i).getSignalPath());
+                }
+            }
+
+        }
+
+        if (path_movementRight == null) {
+            stabVelR = 0;
+        } else {
+            CSVFileReader.Signal TremorSignalaX = FileReader.ReadMovementSignal(path_movementRight, "aX [m/s^2]");
+            CSVFileReader.Signal TremorSignalaY = FileReader.ReadMovementSignal(path_movementRight, "aY [m/s^2]");
+            CSVFileReader.Signal TremorSignalaZ = FileReader.ReadMovementSignal(path_movementRight, "aZ [m/s^2]");
+
+            stabVelR = MovementProcessor.ComputeRegularity(TremorSignalaX.Signal, TremorSignalaY.Signal, TremorSignalaZ.Signal);
+        }
+
+        return (float) stabVelR;
+    }
+    // Compute Stability in Rotation
+    // Left Hand
+    private float CalculateStabRotationLeft(){
```

Comment:
same here

what is the difference between all these functions???

use only one function and called it depending on the name of the file